src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```
@@ -9,6 +9,20 @@
 
     @Override
     public User register(User user) {
+        User tmp = storageDao.get(user.getLogin());
+        if (user.getAge() == null || user.getLogin() == null || user.getPassword() == null) {
+            throw new NullPointerException();
+        }
+
+        if (tmp == null) {
+            if (user.getAge() >= 18 && user.getLogin().length() >= 6 && user.getPassword().length() >= 6) {
+                storageDao.add(user);
+                return user;
+            }
+        } else {
+            return new User();
```

Issue: When a user with the same login already exists, you are returning a new empty `User` object. According to the requirements and checklist, you should throw an exception (such as `IllegalArgumentException` or a custom exception) instead of returning an object. Please update this logic.